tests/tests.test.ts

```diff
@@ -2195,7 +2195,7 @@ describe('Promise values', () => {
     test('when callback works with promises', async () => {
         let resolver: (value: number) => void;
         const promise = new Promise<number>((resolve) => (resolver = resolve));
-        const obs = observable<number>(promise);
+        const obs = observable(promise);
```

Comment:
passing the generic type explicitly would have to be done like so: `observable<Promise<number>>(promise);`

we could make it work with `observable<number>(promise);` by adding two function signature overloads to explicitly handle this scenario:
```typescript
export function observable<T>(value: Promise<T>): MaybePromiseObservable<Promise<T>>;
export function observable<T>(value?: Promise<T>): MaybePromiseObservable<Promise<T> | undefined>;
```

not sure if we want that though. For example, in react this would be a type error:
```typescript
useState<number>(Promise.resolve(1));
```

because of `MaybePromise` utility type, the observable still has the correct type, so that:
```typescript
const obs1 = observable(Promise.resolve(10)); // Observable<number & PromiseInfo>
const obs2 = observable(10)); // Observable<number>

Reply:
This seems good for now. Let's go more in depth into promises in a follow-up PR based on what we discussed in Slack.